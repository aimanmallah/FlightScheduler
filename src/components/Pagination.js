const Pagination = ({ handleNext, handlePrevious, pageNumber }) => {
  return (
    <div className="flex justify-center space-x-8 my-6">
      {pageNumber === 0 ? null : (
        <button
          className="bg-blue-300 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded w-28"
          onClick={() => handlePrevious()}
        >
          Previous
        </button>
      )}
      <p className="text-3xl text-center">{pageNumber}</p>
      <button
        className="bg-blue-300 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded w-28"
        onClick={() => handleNext()}
      >
        Next
      </button>
    </div>
  );
};

export default Pagination;
