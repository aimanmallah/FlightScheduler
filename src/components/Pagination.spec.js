import { fireEvent, render } from '../utils/test-utils';

import Pagination from './Pagination';

const paginationData = {
  handleNext: jest.fn(),
  handlePrevious: jest.fn(),
  pageNumber: 5,
};

describe('Pagination', () => {
  it('renders as expected', () => {
    const { container, getAllByRole, getByText } = render(
      <Pagination pageNumber={paginationData.pageNumber} />
    );

    expect(container.firstChild).toMatchSnapshot();
    expect(getByText(paginationData.pageNumber)).toBeTruthy();
    expect(getAllByRole('button')).toHaveLength(2);
    expect(getByText('Next')).toBeTruthy();
    expect(getByText('Previous')).toBeTruthy();
  });

  it('does not render Previous button when pageNumber equals zero', () => {
    const { queryByText } = render(<Pagination pageNumber={0} />);

    expect(queryByText('Previous')).toBeFalsy();
  });

  it('calls handleNext when Next is clicked', () => {
    const { getByText } = render(
      <Pagination
        pageNumber={paginationData.pageNumber}
        handleNext={paginationData.handleNext}
        handlePrevious={paginationData.handlePrevious}
      />
    );

    const node = getByText('Next');

    fireEvent.click(node);

    expect(paginationData.handleNext).toHaveBeenCalled();
    expect(paginationData.handlePrevious).not.toHaveBeenCalled();
  });

  it('calls handleNext when Previous is clicked', () => {
    const { getByText } = render(
      <Pagination
        pageNumber={paginationData.pageNumber}
        handleNext={paginationData.handleNext}
        handlePrevious={paginationData.handlePrevious}
      />
    );

    const node = getByText('Previous');

    fireEvent.click(node);

    expect(paginationData.handleNext).not.toHaveBeenCalled();
    expect(paginationData.handlePrevious).toHaveBeenCalled();
  });
});
