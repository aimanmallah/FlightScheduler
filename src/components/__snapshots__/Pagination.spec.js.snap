// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders as expected 1`] = `
<div
  class="flex justify-center space-x-8 my-6"
>
  <button
    class="bg-blue-300 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded w-28"
  >
    Previous
  </button>
  <p
    class="text-3xl text-center"
  >
    5
  </p>
  <button
    class="bg-blue-300 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded w-28"
  >
    Next
  </button>
</div>
`;
